WorkflowDemos/WorkflowDemosTest/src_test/test/bpm/TestProcurementRequest.java

```diff
@@ -0,0 +1,225 @@
+    List<ITask> verifyTasks = createProcurrementRequest(bpmClient);
+
+    ui.mock(new PID("15254DC87A1B183B-f5"), ProcurementRequest.class, (in, out) -> out.setDataOkTeamLeader(false));
+    verifyRequest(bpmClient, verifyTasks.get(0), "Teamleader");
+    
+    ui.mock(new PID("15254DC87A1B183B-f6"), ProcurementRequest.class, (in, out) -> out.setDataOkManager(true));
+    verifyRequest(bpmClient, verifyTasks.get(1), "Manager");
+    
+    ICase caze = verifyTasks.get(0).getCase();
+    waitUntilSystemTaskHasBeenExecuted(caze); 
+    
+    assertThat(caze.getState()).isEqualTo(CaseState.DONE);
+  }
+
+ /**
+  * Manager and Team Leader do verify but Executive Manager does not accept 
+  */
+  @Test
+  void approvalTask_verified_notAccepted(BpmClient bpmClient, UiMocker ui) throws Exception
+  {
+    ui.mock(new PID("15254DC87A1B183B-f3"), in -> testData);
+    List<ITask> verifyTasks = createProcurrementRequest(bpmClient);
+
+    ui.mock(new PID("15254DC87A1B183B-f5"), ProcurementRequest.class, (in, out) -> out.setDataOkTeamLeader(true));
+    verifyRequest(bpmClient, verifyTasks.get(0), "Teamleader");
+
+    ui.mock(new PID("15254DC87A1B183B-f6"), ProcurementRequest.class, (in, out) -> out.setDataOkManager(true));
+    verifyRequest(bpmClient, verifyTasks.get(1), "Manager");
+    
+    ICase caze = verifyTasks.get(0).getCase();
+    waitUntilAcceptRequestTaskIsAvailable(caze);
+    
+    ui.mock(new PID("15254DC87A1B183B-f16"));
+    ProcurementRequest request = acceptRequest(bpmClient, getAcceptRequestTask(caze));
+    
+    assertThat(request.getAccepted()).isFalse();
+    assertThat(caze.getState()).isEqualTo(CaseState.DONE);
+  }
+  
+  /**
+   * Manager and Team Leader do verify and Executive Manager does accept 
+   */
+   @Test
+   void approvalTask_verified_accepted(BpmClient bpmClient, UiMocker ui) throws Exception
+   {
+     ui.mock(new PID("15254DC87A1B183B-f3"), in -> testData);
+     List<ITask> verifyTasks = createProcurrementRequest(bpmClient);
+
+     ui.mock(new PID("15254DC87A1B183B-f5"), ProcurementRequest.class, (in, out) -> out.setDataOkTeamLeader(true));
+     verifyRequest(bpmClient, verifyTasks.get(0), "Teamleader");
+
+     ui.mock(new PID("15254DC87A1B183B-f6"), ProcurementRequest.class, (in, out) -> out.setDataOkManager(true));
+     verifyRequest(bpmClient, verifyTasks.get(1), "Manager");
+
+     ICase caze = verifyTasks.get(0).getCase();
+     waitUntilAcceptRequestTaskIsAvailable(caze);
+     
+     ui.mock(new PID("15254DC87A1B183B-f16"), ProcurementRequest.class, (in, out) -> out.setAccepted(true));
+     ProcurementRequest request = acceptRequest(bpmClient, getAcceptRequestTask(caze));
+     
+     assertThat(request.getAccepted()).isTrue();
+     assertThat(caze.getState()).isEqualTo(CaseState.DONE);
+   }
+
+  private List<ITask> createProcurrementRequest(BpmClient bpmClient) throws Exception
+  {
+    ExecutionResult result = bpmClient
+        .start().process(Process.ProcurementRequest+"/start.ivp")
```

Comment:
we no longer need this 'internal GUID path driven' path. Better go for the human readable alternative ...without even using 'start.ivp'